libs/wingc/src/jsify.rs

```diff
@@ -227,10 +227,71 @@ impl<'a> JSifier<'a> {
 		}
 	}
 
-	fn jsify_type(&self, typ: &TypeAnnotationKind) -> String {
+	fn jsify_type(&self, typ: &TypeAnnotationKind) -> Option<String> {
 		match typ {
-			TypeAnnotationKind::UserDefined(t) => self.jsify_user_defined_type(&t),
-			_ => todo!(),
+			TypeAnnotationKind::UserDefined(t) => Some(self.jsify_user_defined_type(&t)),
+			TypeAnnotationKind::String => Some("string".to_string()),
+			TypeAnnotationKind::Number => Some("number".to_string()),
+			TypeAnnotationKind::Bool => Some("boolean".to_string()),
+			TypeAnnotationKind::Array(t) => {
+				if let Some(inner) = self.jsify_type(&t.kind) {
+					Some(format!("{}[]", inner))
+				} else {
+					None
+				}
+			}
+			TypeAnnotationKind::Optional(t) => {
+				if let Some(inner) = self.jsify_type(&t.kind) {
+					Some(format!("{}?", inner))
+				} else {
+					None
+				}
+			}
+			_ => None,
+		}
+	}
+
+	// This helper determines what requirement and dependency to add to the struct schema based
+	// on the type annotation of the field.
+	// I.E. if a struct has a field named "foo" with a type "OtherStruct", then we want to add
+	// the field "foo" as a required  and the struct "OtherStruct" as a dependency. so the result is
+	// a tuple (required, dependency)
+	fn extract_struct_schema_dependencies(
```

Comment:
Maybe `extrac_struct_field_schema_dependency` (note the added field, because we're dealing with a single field and removal of the plural "dependencies" because we're looking for a single dependency for a field).